src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -3855,75 +3856,132 @@ private void BuildComplexPropertyValue(IIndentedStringBuilder writer, XamlMember
 		{
 			TryAnnotateWithGeneratorSource(writer);
 			Func<string, string> formatLine = format => prefix + format + (prefix.HasValue() ? ";\r\n" : "");
+			var postfix = prefix.HasValue() ? ";" : "";
 
 			var bindingNode = member.Objects.FirstOrDefault(o => o.Type.Name == "Binding");
 			var bindNode = member.Objects.FirstOrDefault(o => o.Type.Name == "Bind");
 			var templateBindingNode = member.Objects.FirstOrDefault(o => o.Type.Name == "TemplateBinding");
 
-			string? GetBindingOptions()
-			{
-				if (bindingNode != null)
-				{
-					return bindingNode
-						.Members
-						.Select(BuildMemberPropertyValue)
-						.JoinBy(", ");
-
-				}
-				if (bindNode != null)
-				{
-					return bindNode
-						.Members
-						.Where(m => m.Member.Name != "_PositionalParameters" && m.Member.Name != "Path" && m.Member.Name != "BindBack")
-						.Select(BuildMemberPropertyValue)
-						.Concat(bindNode.Members.Any(m => m.Member.Name == "Mode") ? "" : "Mode = BindingMode." + GetDefaultBindMode())
-						.JoinBy(", ");
-
-				}
-				if (templateBindingNode != null)
-				{
-					return templateBindingNode
-						.Members
-						.Select(BuildMemberPropertyValue)
-   						.Concat("RelativeSource = new RelativeSource(RelativeSourceMode.TemplatedParent)")
-					 .JoinBy(", ");
-				}
-
-				return null;
-			}
-
 			if (FindEventType(member.Member) is IEventSymbol eventSymbol)
 			{
 				GenerateInlineEvent(closureName, writer, member, eventSymbol);
 			}
 			else
 			{
-				var bindingOptions = GetBindingOptions();
-
-				if (bindingOptions != null)
+				if ((bindingNode ?? templateBindingNode ?? templateBindingNode) != null)
```

Comment:
this & the top part of WriteBinding smells... try to extract the result as an `func<>?` and restore the old pattern